src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -80,8 +81,11 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
   const fetchDataSources = () => {
     setIsLoading(true);
     getDataSources(savedObjects.client)
-      .then((response: DataSourceTableItem[]) => {
-        setDataSources(response);
+      .then(async (response: DataSourceTableItem[]) => {
+        return await fetchDataSourceConnections(response, http, notifications);
+      })
```

Comment:
Any reason for not simply chaining the promise?
```suggestion
      .then((response: DataSourceTableItem[]) =>
        fetchDataSourceConnections(response, http, notifications);
      )
```